openstack_terraforming/compute.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2018 The Terraformer Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package openstack_terraforming
+
+import (
+	"strings"
+
+	"github.com/GoogleCloudPlatform/terraformer/terraform_utils"
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack"
+	"github.com/gophercloud/gophercloud/openstack/compute/v2/servers"
+	"github.com/gophercloud/gophercloud/pagination"
+)
+
+type ComputeGenerator struct {
+	OpenStackService
+}
+
+// createResources iterate on all openstack_compute_instance_v2
+func (g *ComputeGenerator) createResources(list *pagination.Pager) []terraform_utils.Resource {
+	resources := []terraform_utils.Resource{}
+
+	list.EachPage(func(page pagination.Page) (bool, error) {
+		servers, err := servers.ExtractServers(page)
+		if err != nil {
+			return false, err
+		}
+
+		for _, s := range servers {
+			resource := terraform_utils.NewResource(
+				s.ID,
+				s.Name,
+				"openstack_compute_instance_v2",
+				"openstack",
+				map[string]string{
+					"name": s.Name,
```

Comment:
I think no needed "name", openstack provider use only ID for extract data
https://github.com/terraform-providers/terraform-provider-openstack/blob/master/openstack/resource_openstack_compute_instance_v2.go#L548